tx-bakery/src/lib.rs

```diff
@@ -924,15 +951,24 @@ impl TxBakery {
             .map(|r| TxBakery::apply_ex_units(r, &ex_units))
             .collect::<Result<Vec<_>>>()?;
 
-        let tx_body = self.balance_transaction(&tx, tx_builder, &datums, &redeemers_w_ex_u)?;
+        let tx_body = self.balance_transaction(
+            &tx,
+            tx_builder,
+            &datums,
+            &redeemers_w_ex_u,
+            &wit_scripts,
+            &ref_scripts,
+        )?;
 
         let mut witness_set = csl::TransactionWitnessSet::new();
         let mut script_witnesses = csl::PlutusScripts::new();
 
-        plutus_scripts
+        wit_scripts
             .iter()
             .for_each(|script| script_witnesses.add(script));
 
+        // let ref_scripts = tx.tx_info.reference_inputs.iter().any(|tx_in_info| tx_in_info.output.reference_script.is_so);
+
         let mut redeemer_witnesses = csl::Redeemers::new();
```

Comment:
```suggestion
        let mut redeemer_witnesses = csl::Redeemers::new();
```

Probably should just kill these lines (the commented lines) if we don't need it -- up to you in the end though!

Reply:
oops, you're right